SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialAuthorityTest/SpatialAuthorityTest.cpp

```diff
@@ -0,0 +1,387 @@
+				Timer -= DeltaTime;
+				if (Timer <= 0)
+				{
+					const FWorkerDefinition& LocalWorkerDefinition = GetLocalFlowController()->WorkerDefinition;
+					if (LocalWorkerDefinition.Type == ESpatialFunctionalTestWorkerType::Server && LocalWorkerDefinition.Id == 1)
+					{
+						if (VerifyTestActor(LevelReplicatedActor, 1, 1, 1, 0))
+						{
+							FinishStep();
+						}
+					}
+					else
+					{
+					
+						if (VerifyTestActor(LevelReplicatedActor, 0, 0, 0, 0))
+						{
+							FinishStep();
+						}
+					}
+				}
+			}, 5.0f);
+	}
+
+	// Non-replicated Level Actor. Each Server should have Authority over their instance, Clients don't.
+	{
+		AddStep(TEXT("Non-replicated Level Actor - Each Server has Authority, Client doesn't know"), FWorkerDefinition::AllWorkers, nullptr, nullptr,
+			[this](float DeltaTime) {
+				// Since this actor already was in level and we wait for timer in the previous step, we don't need to wait
+				// in this one again.
+				const FWorkerDefinition& LocalWorkerDefinition = GetLocalFlowController()->WorkerDefinition;
+				if (LocalWorkerDefinition.Type == ESpatialFunctionalTestWorkerType::Server)
+				{
+					// Note: Non-replicated actors never get OnAuthorityGained() called.
+					if (VerifyTestActor(LevelActor, LocalWorkerDefinition.Id, LocalWorkerDefinition.Id, 0, 0))
+					{
+						FinishStep();
+					}
+				}
+				else
+				{
+					if (VerifyTestActor(LevelActor, 0, 0, 0, 0))
+					{
+						FinishStep(); // Clients don't have authority over non-replicated Level Actors.
+					}
+				}
+			}, 5.0f);
+	}
+
+	// Replicated Dynamic Actor Spawned On Same Server. Server 1 should have Authority.
+	{
+		AddStep(TEXT("Replicated Dynamic Actor Spawned On Same Server - Spawn"), FWorkerDefinition::Server(1), nullptr,
+			[this]()
+			{
+				ASpatialAuthorityTestReplicatedActor* Actor = GetWorld()->SpawnActor<ASpatialAuthorityTestReplicatedActor>(
+					Server1Position, FRotator::ZeroRotator);
+				CrossServerSetDynamicReplicatedActor(Actor);
+				FinishStep();
+			});
+
+		AddStep(TEXT("Replicated Dynamic Actor Spawned On Same Server - Verify Server 1 Has Authority"), FWorkerDefinition::AllWorkers, nullptr, nullptr,
+			[this](float DeltaTime)
+			{
+				Timer -= DeltaTime;
+				if (Timer <= 0)
+				{
+					const FWorkerDefinition& LocalWorkerDefinition = GetLocalFlowController()->WorkerDefinition;
+					if( LocalWorkerDefinition.Type == ESpatialFunctionalTestWorkerType::Server && LocalWorkerDefinition.Id == 1)
+					{
+						if (VerifyTestActor(DynamicReplicatedActor, 1, 1, 1, 0))
+						{
+							FinishStep();
+						}
+					}
+					else
+					{
+						if (VerifyTestActor(DynamicReplicatedActor, 0, 0, 0, 0))
+						{
+							FinishStep();
+						}
+					}
+				}
+			}, 5.0f);
+
+		AddStep(TEXT("Replicated Dynamic Actor Spawned On Same Server - Destroy"), FWorkerDefinition::Server(1), nullptr,
+			[this]()
+			{
+				DynamicReplicatedActor->Destroy();
+				CrossServerSetDynamicReplicatedActor(nullptr);
```

Comment:
Should be redundant, but I guess there's no harm

Reply:
I like to force things to null, it's a habit specially of using these engines were things are no longer valid but are not null.